game/navigator/navigator_service.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 var ns *navService
```

Comment:
Avoid globals. I know the stdlib relies on it too much as well, do not take the gods of golang for granted on this.
Take a look at [wire](https://github.com/google/wire) and [this](https://blog.drewolson.org/go-dependency-injection-with-wire) explanatory blog post for a better strategy.

Reply:
So this is probably my biggest hold up right now. I have never made it much further than where I am right now in making a Habbo server and so my knowledge of how things like items and room instances are handled is limited to some extent. 

Thinking about it initially it seemed like we would need to be able to access the various services all over the application. My naive idea at first was to start the services in `main` and then pass them into the `Server` struct where a `Player` could call them via their underlying `Session`. Given most Habbo servers I have looked at in detail use this global singleton pattern I figured maybe this was the only way to go about it but maybe further thought needs to be put in here because yeah global state is not good and imo singletons are an anti-pattern.

Maybe some synchronous discussions between you and I could be beneficial because I know you have far more context/insight into these things.